tests/inference/ebm_test.py

```diff
@@ -26,99 +27,137 @@
 from tests import test_util
 
 
-class NullEnergy(models.BitstringEnergy):
-  """Simple empty energy."""
-
-  def __init__(self, bits):
-    """Initializes a NullEnergy."""
-    energy_layers = []
-    super().__init__(bits, energy_layers)
-
-
 class EnergyInferenceTest(tf.test.TestCase):
   """Tests a simple instantiation of EnergyInference."""
 
-  class TwoOutcomes(inference.EnergyInference):
-    """EnergyInference which is independent of the input energy."""
-
-    def __init__(self, null_energy, num_expectation_samples, bitstring_1,
-                 bitstring_2, p_1):
-      """Initializes a simple inference class.
-
-      Args:
-        null_energy: Energy function that will be ignored.
-        num_expectation_samples: Number of samples to draw and use for
-          estimating the expectation value.
-        bitstring_1: First bitstring to sample.
-        bitstring_2: Second bitstring to sample.
-        p_1: probability of sampling the first bitstring.
-      """
-      super().__init__(null_energy, num_expectation_samples)
-      self.bitstring_1 = bitstring_1
-      self.bitstring_2 = bitstring_2
-      self.p_1 = p_1
+  class EnergyInferenceBernoulliSampler(inference.EnergyInference):
```

Comment:
jfme: why not subclass `AnalyticEnergyInference`?

Reply:
This way the sample based `entropy` and `log_partition` functions get tested, rather than the analytic version from `AnalyticEnergyInference`

Reply:
Whoops yeah makes sense - noticed that `_ready_inference` looks the same which made me wonder. SG.